### ISSUE TITLE
multi-threaded distance_array()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [MDAnalysis/mdanalysis#80]
```
It would be nice to have a multi-threaded version of the distance_array() and 
self_distance_array() functions to speed up this time consuming calculation.

A possible starting point could be 

  Multi-threaded vectorized distance matrix computation on the CELL/BE and x86/SSE2 architectures. Adrianto Wirawan, Chee Keong Kwoh and Bertil Schmidt. Bioinformatics (2010) 26 (10): 1368-1369. doi: 0.1093/bioinformatics/btq135 

  http://bioinformatics.oxfordjournals.org/content/26/10/1368.full

Code is available at https://sourceforge.net/projects/distmatcomp/

Anyone wants to take this on as a nice little project in parallel programming? 
:-)


```

Original issue reported on code.google.com by `orbeckst` on 1 Oct 2011 at 9:15
